projecttacoma/fqm-testify
# Add QI-Core profiles to resources and patients

# Summary
This PR uses the addition of profiles on the dataRequirements output in `fqm-execution` in [this](https://github.com/projecttacoma/fqm-execution/pull/284) PR on resources and patients.

## New Behavior
QI-Core resources and patients now have QI-Core `meta.profile` attributes on them. 

## Code Changes
- `resourceCreation.ts` - update `createPatientResourceString` function to add qi-core `meta.profile` if the patient is a qi-core patient and update `createFHIRResourceString` function to add qi-core `meta.profile` for the resource if it is a qi-core resource.
- `PatientCreationPanel.tsx` - goes through the dataRequirements and if any of them have a qi-core profile, then pass true into the call to `createPatientReosurceString` function. Question- is there a better way to do this? Is there a better way to check if a patient should have a qi-core `meta.profile`?
- `PatientCreationPanel.test.tsx` - these were the unit tests I was fighting with. With the addition of using the `dataRequirements` selector in `PatientCreationPanel`, I got warnings so I had to add `Suspense` and `async / act` to the unit tests (like how it's done in `DataRequirementsSelectOption.test.tsx` and `ResourceSelection.test.tsx`). I also added a check that qicore-patient `meta.profile` gets added to the patient template when the dataRequirements contains a qi-core profile.
- `TestCaseEditor.test.tsx` - added `Suspense` and `async / act` for the same reason as above.

# Testing Guidance
- `npm run check`
- Use npm link to link this branch to the `qicore-profiles` branch of `fqm-execution`.
- Upload a measure bundle that was authored using QI-Core (any of the CMS measures).
- Create a patient. Make sure the patient JSON includes `meta.profile`.
- Add some resources to the patient. Make sure these resources include the proper `meta.profile` as well.
- Take a close look at the unit tests. Do the changes I made make sense? Is there a better way to check if a patient should have a qi-core `meta.profile`?
